### DISCUSSION TITLE
Shadow endpoints can't set cookies and redirect, but standalone endpoints can - is this a bug?

### DISCUSSION THREAD

**Discussion by goynang** [sveltejs/kit#3739] [Q&A] [answered]
I have some code in a shadow endpoint that does some logic and either renders the matching page or sets a cookie and redirects elsewhere.

This worked fine in older versions of Kit but since shadow endpoints have become a thing it no longer sets the cookie. If I rename the endpoint so that it is not a shadow one then it works (i.e. sets the cookie) but this seems less than elegant as in some cases I still want to fall through to the previously matching .svelte route. I can't see a reason why it would be OK to set cookies on standalone endpoints but not shadow ones.

From munging tests within kit itself I can see that **this behavour only occurs when redirecting** now - the cookie is set fine if you just fall through.

Simplified version of my redirect code in the endpoint...

````javascript
return {
    status: 303,
    headers: {
        'set-cookie': 'foo=bar; HttpOnly',
        location: '/'
    }
}
````

Am I doing this wrong, or should this work?

**Comment by Mlocik97** [comment]
yeah, I think it is...

**Comment by georgecrawford** [comment, marked as answer]
This seems like the same issue I reported: https://github.com/sveltejs/kit/issues/3758

**Reply by goynang** [reply]
Yep - it sure does.  I won't convert this discussion into an issue but will take what ever answer comes from your issue instead.

**Reply by georgecrawford** [reply]
https://github.com/sveltejs/kit/issues/3758 is now fixed and released!

**Reply by georgecrawford** [reply]
can you mark this as answered?
